h/storage.py

```diff
@@ -180,18 +202,20 @@ def update_annotation(request, id_, data):
             "group: " + _("Invalid group specified for annotation")
         )
 
-    if document:
+    if document or uri_changed:
```

Comment:
This is a fix for a long-standing bug where an annotation's `document_id` did not get updated if an API caller called `POST /api/annotations` to update an annotation's URI, but didn't provide any document metadata.

Reply:
Is this an unrelated change? Could it be moved to a separate PR?

Reply:
This change is needed in case you try to use the URL migration tool and only provide a new URL with no document metadata, it also fixes a bug which has long existed but probably has only been run into by third-parties before now so we hadn't noticed.